### PR TITLE
SPCTR-2359 - Implemented isSelect and refactor useEffect

### PR DESCRIPTION
**Pull request by rahulvarma722**
1. Remove react and start using wp.element.
2. Preview image sifted to the block.js previously used in edit.js.
3. Created a preview image component in the preview image component you need to pass the image slug and isChildren.
4. For setting components use block props property isSelected in condition if isSelected is true then the component should mount.

### CODE REVIEW

**Review by patilvikasj — APPROVED** [review]

**Review by supriyajjadhav — APPROVED** [review]

**Review by aapatre — APPROVED** [review]
LGTM!

**Review by supriyajjadhav — APPROVED** [review]

File: src/blocks/wp-search/edit.js [hunk 1]
```diff
@@ -98,15 +99,11 @@ const UAGBWpSearchEdit = ( props ) => {
...
-
 	return (
-		props.attributes.isPreview ? <img width='100%' src={ previewImageData } alt=''/> : (
-			<>
-				<Settings parentProps={ props } />
-				<Render parentProps={ props } />
-			</>
-		)
+		<>
+		{ isSelected && <Settings parentProps={ props } /> }
```
Comment (adi3890) [inline src/blocks/wp-search/edit.js:104]: please check indentation

File: src/components/responsive-image/index.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
 /**
  * External dependencies
  */
-import React, {useEffect, useState, useRef } from 'react';
+import { useEffect, useState,useRef } from '@wordpress/element';
```
Comment (adi3890) [inline src/components/responsive-image/index.js:4]:
```suggestion
import { useEffect, useState, useRef } from '@wordpress/element';
```

File: src/components/responsive-icons/device-icons.js [hunk 3]
```diff
@@ -4,10 +4,9 @@
 import { ButtonGroup, Button, Tooltip  } from '@wordpress/components';
 import { useDeviceType } from '@Controls/getPreviewType';
 import { __, sprintf } from '@wordpress/i18n'
-import { useCallback } from '@wordpress/element'
+import { useCallback,useLayoutEffect } from '@wordpress/element'
```
Comment (adi3890) [inline src/components/responsive-icons/device-icons.js:7]:
```suggestion
import { useCallback, useLayoutEffect } from '@wordpress/element'
```

File: src/components/reset/index.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import React, { useEffect, useState, useRef } from 'react';
+import {  useEffect, useState, useRef } from '@wordpress/element';
```
Comment (adi3890) [inline src/components/reset/index.js:1]:
```suggestion
import { useEffect, useState, useRef } from '@wordpress/element';
```

File: src/components/presets/index.js [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import { SelectControl } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import styles from './editor.lazy.scss';
-import React, { useLayoutEffect, useEffect, useState, useRef } from 'react';
+import { useLayoutEffect, useEffect, useState, useRef,memo } from '@wordpress/element';
```
Comment (adi3890) [inline src/components/presets/index.js:4]:
```suggestion
import { useLayoutEffect, useEffect, useState, useRef, memo } from '@wordpress/element';
```
avoid adding extra line

File: src/components/multi-buttons-control/index.js [hunk 6]
```diff
@@ -1,7 +1,8 @@
 /**
  * WordPress dependencies
  */
-import React, { useLayoutEffect, useEffect, useState, useRef } from 'react';
+import { useLayoutEffect,useEffect, useState, useRef } from '@wordpress/element';
```
Comment (adi3890) [inline src/components/multi-buttons-control/index.js:4]:
```suggestion
import { useLayoutEffect, useEffect, useState, useRef } from '@wordpress/element';
```

File: src/components/image/index.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import React, {useEffect, useState, useRef } from 'react';
+import { useEffect, useState,useRef } from '@wordpress/element';
```
Comment (adi3890) [inline src/components/image/index.js:1]:
```suggestion
import { useEffect, useState, useRef } from '@wordpress/element';
```

**Comment by adi3890** [comment]
@rahul i could see multiple indentation, spacing issues in js can you please create a JIRA task and try to setup prettier to resolve it.

File: src/blocks/wp-search/edit.js (hunk 1 above)
Comment (rahulvarma722) [reply src/blocks/wp-search/edit.js:104]: There are lots of indentation inconsistency in this plugin so in next to next pr we will implement prettier so these types of indentation will be resolved

File: blocks-config/uagb-controls/previewImage.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+const PreviewImage = ( { image, isChildren = false } ) => {
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply blocks-config/uagb-controls/previewImage.js:1]: yes

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**